lib/spack/spack/util/environment.py

```diff
@@ -767,9 +767,7 @@ def from_sourcing_file(
 
         # Compute the environments before and after sourcing
         before = sanitize(
-            environment_after_sourcing_files(os.devnull, **kwargs),
-            exclude=exclude,
-            include=include,
+            environment_after_sourcing_files("cd %CD%", **kwargs), exclude=exclude, include=include
```

Comment:
This would only ever work on Windows and will break cross platform support. We should replace this with a variable that contains a cross platform solution.